### PR TITLE
Add tests for OpenSshProtocolCheck

### PR DESCRIPTION
**Pull request by drehak**
- split functionality into the library
- add tests

Heavily similar to #424, since they consume the same model.

### CODE REVIEW

**Comment by drehak** [comment]
@Jakuje

File: repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+def test_no_config(current_actor_context):
+    with pytest.raises(StopActorExecutionError):
+        library.process(iter([]))
+
+
+osprl = OpenSshPermitRootLogin(value='no')
+
+
+@pytest.mark.parametrize('protocol', ['', '7'])
```
Comment (Jakuje) [inline repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py]: Just a nit that the real values that could be in the config are `1`, `2` or `1,2` so testing these would probably be cleaner, but this does its job fine.

Comment (drehak) [reply repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py]: I see... Updated the tests, thanks.

File: repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    with pytest.raises(StopActorExecutionError):
+        library.process(iter([]))
+
+
+osprl = OpenSshPermitRootLogin(value='no')
+
+
+@pytest.mark.parametrize('protocol', ['', '1', '2', '1,2', '2,1', '7'])
+def test_protocol(current_actor_context, protocol):
+    if protocol:
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py]: Nit: I suspect L19-27 can be squashed into `current_actor_context.feed(OpenSshConfig(permit_root_login=[osprl], protocol=protocol))` if no_protocol test case is passed as None.

Comment (drehak) [reply repos/system_upgrade/el7toel8/actors/opensshprotocolcheck/tests/unit_test.py]: Yeah, thanks.

**Comment by drehak** [comment]
Applied feedback and rebased.

**Pull request merged**